cli/loader/src/lib.rs

```diff
@@ -1112,30 +1113,32 @@ impl Loader {
     ) -> Result<&[LanguageConfiguration]> {
         let initial_language_configuration_count = self.language_configurations.len();
 
-        if let Some(config) = TreeSitterJSON::from_file(parser_path) {
-            let language_count = self.languages_by_id.len();
-            for grammar in config.grammars {
-                // Determine the path to the parser directory. This can be specified in
-                // the tree-sitter.json, but defaults to the directory containing the
-                // tree-sitter.json.
-                let language_path = parser_path.join(grammar.path.unwrap_or(PathBuf::from(".")));
-
-                // Determine if a previous language configuration in this package.json file
-                // already uses the same language.
-                let mut language_id = None;
-                for (id, (path, _, _)) in
-                    self.languages_by_id.iter().enumerate().skip(language_count)
-                {
-                    if language_path == *path {
-                        language_id = Some(id);
+        match TreeSitterJSON::from_file(parser_path) {
```

Comment:
`if let Ok(...) { ... } else {...}` instead please.

Reply:
I tried assigning the result of the call to a temp and destructuring it in `if let`/ `else if let` blocks. Is there a way to access the `Err` value more directly in the `else` block of an `if let`?